ir/src/main/scala/org/scalajs/ir/Hashers.scala

```diff
@@ -441,10 +441,6 @@ object Hashers {
           throw new InvalidIRException(tree,
               "Cannot hash a transient IR node (its value is of class " +
               s"${value.getClass})")
-
-        case _ =>
-          throw new IllegalArgumentException(
-              s"Unable to hash tree of class ${tree.getClass}")
```

Comment:
This `match` is in fact provably exhaustive, and Dotty reports a warning for a `case _ =>` when the rest is otherwise exhaustive.